### ISSUE TITLE
--format doesn't respect custom Indentation

### ISSUE THREAD

**Issue by b1zzu** [nvuillam/npm-groovy-lint#31]
Hi,
this is may groovy file:
```groovy
pipeline {
  agent none
  options {
    timestamps()
    ansiColor('xterm')
  }
  stages {
    stage('stuff') {
      steps {
        sh 'echo "stuff"'
      }
    }
  }
}

```
and this is my `.groovylintrc.json` configuration:
```json
{
  "extends": "recommended-jenkinsfile",
  "rules": {
    "Indentation": {
      "spacesPerIndentLevel": 2
    }
  }
}
```
as you can see the groovy file is correct and if I analyze the file it doesn't report any error 

```
➜ npm-groovy-lint         

npm-groovy-lint results in  linted files:
┌─────────┬───────────┬─────────────┐
│ (index) │ Severity  │ Total found │
├─────────┼───────────┼─────────────┤
│    0    │  'Error'  │      0      │
│    1    │ 'Warning' │      0      │
│    2    │  'Info'   │      0      │
└─────────┴───────────┴─────────────┘
```

but if I try to automatically format it will indent everything with four spaces instant of two

```
➜  npm-groovy-lint --format
/home/b1zzu/Projects/gitlab.cee.redhat.com/integreatly-qe/ci-cd/stuff/test.groovy


npm-groovy-lint results in  linted files:
┌─────────┬───────────┬─────────────┬─────────────┬─────────────────┐
│ (index) │ Severity  │ Total found │ Total fixed │ Total remaining │
├─────────┼───────────┼─────────────┼─────────────┼─────────────────┤
│    0    │  'Error'  │      0      │      0      │        0        │
│    1    │ 'Warning' │      0      │      0      │        0        │
│    2    │  'Info'   │      8      │      8      │        0        │
└─────────┴───────────┴─────────────┴─────────────┴─────────────────┘
```

and if I analyze again the modified it will show indentation problems:
```
➜  npm-groovy-lint         
/home/b1zzu/Projects/gitlab.cee.redhat.com/integreatly-qe/ci-cd/stuff/test.groovy
  2     info     The statement on line 2 in class None is at the incorrect indent level: Expected column 3 but was 5  Indentation             
  3     info     The statement on line 3 in class None is at the incorrect indent level: Expected column 3 but was 5  Indentation             
  4     info     The statement on line 4 in class None is at the incorrect indent level: Expected column 5 but was 9  Indentation             
  5     info     The statement on line 5 in class None is at the incorrect indent level: Expected column 5 but was 9  Indentation             
  7     info     The statement on line 7 in class None is at the incorrect indent level: Expected column 3 but was 5  Indentation             
  8     info     The statement on line 8 in class None is at the incorrect indent level: Expected column 5 but was 9  Indentation             
  9     info     The statement on line 9 in class None is at the incorrect indent level: Expected column 7 but was 13  Indentation             
  10    info     The statement on line 10 in class None is at the incorrect indent level: Expected column 9 but was 17  Indentation             


npm-groovy-lint results in  linted files:
┌─────────┬───────────┬─────────────┐
│ (index) │ Severity  │ Total found │
├─────────┼───────────┼─────────────┤
│    0    │  'Error'  │      0      │
│    1    │ 'Warning' │      0      │
│    2    │  'Info'   │      8      │
└─────────┴───────────┴─────────────┘
```

**Comment by nvuillam** [comment]
I plead guilty :p 
When --format is called, [.groovylintrc-format.json](https://github.com/nvuillam/npm-groovy-lint/blob/master/.groovylintrc-format.json) is called, and it ignores the custom .groovylintrc.json
The solution is to merge custom .groovylintrc.json into .groovylintrc-format.json so overrides are taken in account , it could be done by adding a if mode === 'format' then loadConfig(lint) then merge config.rules before returning results,  there :
https://github.com/nvuillam/npm-groovy-lint/blob/7286a29815b8e5717c92ea0d9b5d3bcc5e3ef66e/src/config.js#L62

I'll do that when i have time, or you can make a PR if you''re in a hurry :)

**Comment by b1zzu** [comment]
Hi @nvuillam I will try to give a look in the evening maybe I manage to create the PR

**Comment by nvuillam** [comment]
@b1zzu fixed, please repoen the issue if you still see a problem :) 

( not included in VsCode Groovy Lint yet, some updates may be necessary, i think there is a hardcoded "4" in VsCodeGroovyLint code which repaces tabs by spaces, I have to solve that before releasing a new version )

**Comment by nvuillam** [comment]
@b1zzu Evolution has been included in new VsCode Groovy Lint version 0.9.5 :)
